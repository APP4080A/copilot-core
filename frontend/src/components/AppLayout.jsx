// src/components/AppLayout.jsx
import React from 'react';
import { Outlet } from 'react-router-dom'; // Import Outlet
import Navbar from './Navbar'; // Import your Navbar component

function AppLayout() {
    return (
        <div className="app-layout">
            <Navbar /> {/* The Navbar will always be rendered here */}
            <main className="app-content">
                <Outlet /> {/* This is where the nested route components will be rendered */}
            </main>
        </div>
    );
}

export default AppLayout;