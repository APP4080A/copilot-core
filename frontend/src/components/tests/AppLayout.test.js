import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AppLayout from '../AppLayout';

// A dummy child to simulate <Outlet />
const DummyComponent = () => <div>Child content here</div>;

jest.mock('react-router-dom', () => {
  const actual = jest.requireActual('react-router-dom');
  return {
    ...actual,
    Outlet: () => <DummyComponent />,
  };
});

jest.mock('../Navbar', () => () => <nav>Mock Navbar</nav>);

describe('AppLayout Component', () => {
  test('renders without crashing', () => {
    render(
      <MemoryRouter>
        <AppLayout />
      </MemoryRouter>
    );
    expect(screen.getByText(/Mock Navbar/i)).toBeInTheDocument();
  });

  test('renders Navbar always', () => {
    render(
      <MemoryRouter>
        <AppLayout />
      </MemoryRouter>
    );
    expect(screen.getByRole('navigation')).toBeInTheDocument();
  });

  test('renders child content via Outlet', () => {
    render(
      <MemoryRouter>
        <AppLayout />
      </MemoryRouter>
    );
    expect(screen.getByText(/Child content here/i)).toBeInTheDocument();
  });

  test('has layout styling applied', () => {
    const { container } = render(
      <MemoryRouter>
        <AppLayout />
      </MemoryRouter>
    );
    expect(container.firstChild).toHaveStyle('min-width: 1200px');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AppLayout />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
