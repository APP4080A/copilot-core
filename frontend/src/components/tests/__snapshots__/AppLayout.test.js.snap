// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-layout"
    style="min-width: 1200px;"
  >
    <nav>
      Mock Navbar
    </nav>
     
    <main
      class="app-content"
    >
      <div>
        Child content here
      </div>
       
    </main>
  </div>
</DocumentFragment>
`;
